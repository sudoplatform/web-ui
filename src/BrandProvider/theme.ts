import { DefaultTheme } from 'styled-components'

const defaultColors = {
  primary: '#3770f6',
  success: '#1ccd82',
  error: '#ff403d',
  interactive: '#3770f6',
}

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof defaultColors
    grays: {
      gray5: string
      gray10: string
      gray20: string
      gray40: string
      gray90: string
    }
    spacing: {
      small: string
      medium: string
      large: string
    }
    padding: {
      small: string
      medium: string
      large: string
    }
    borderRadius: {
      small: string
      medium: string
      large: string
    }
    borderWidth: {
      light: string
      medium: string
      heavy: string
    }
    fontSize: {
      body: string
      h1: string
      h2: string
      h3: string
    }
  }
}

export type BrandColors = typeof defaultColors

export const defaultTheme: DefaultTheme = {
  colors: defaultColors,
  grays: {
    gray5: '#f5f5f7',
    gray10: '#dde1e6',
    gray20: '#c0c9d4',
    gray40: '#656d7f',
    gray90: '#0f1111',
  },
  spacing: {
    small: '6px',
    medium: '10px',
    large: '20px',
  },
  padding: {
    small: '6px',
    medium: '10px',
    large: '20px',
  },
  borderRadius: {
    small: '4px',
    medium: '8px',
    large: '12px',
  },
  borderWidth: {
    light: '0.5px',
    medium: '1px',
    heavy: '2px',
  },
  fontSize: {
    body: '12px',
    h1: '24px',
    h2: '18px',
    h3: '15px',
  },
}
