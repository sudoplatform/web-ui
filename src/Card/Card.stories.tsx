import { storiesOf } from '@storybook/react'
import React from 'react'

import { Card } from './Card'

storiesOf('Card', module).add('Card', () => <Card title="Title">Content</Card>)
