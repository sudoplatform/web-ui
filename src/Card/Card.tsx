import React from 'react'
import styled from 'styled-components'

const Container = styled.div`
  width: 100%;
  border-radius: ${({ theme }) => theme.borderRadius.small};
  box-shadow: rgba(0, 0, 0, 0.12) 0px 2px 10px 0px;
  padding: ${({ theme }) => theme.padding.large};
  background: white;
`

const Title = styled.h2`
  font-size: ${({ theme }) => theme.fontSize.h2};
`

interface Props {
  title?: React.ReactNode
  className?: string
  children?: React.ReactNode
}

export const Card: React.FC<Props> = (props) => (
  <Container className={props.className}>
    {props.title && <Title>{props.title}</Title>}
    {props.children}
  </Container>
)
