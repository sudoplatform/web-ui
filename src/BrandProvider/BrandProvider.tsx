import 'antd/dist/antd.min.css'

import React from 'react'
import { ThemeProvider } from 'styled-components'

import { BrandColors, defaultTheme } from './theme'

interface Props {
  colors?: BrandColors
}

export const BrandProvider: React.FC<Props> = (props) => {
  const theme = {
    ...defaultTheme,
    colors: {
      ...defaultTheme.colors,
      ...props.colors,
    },
  }

  return <ThemeProvider theme={theme}>{props.children}</ThemeProvider>
}
