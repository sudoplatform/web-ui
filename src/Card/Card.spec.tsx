import React from 'react'
import { create } from 'react-test-renderer'

import { BrandProvider } from '../BrandProvider'
import { Card } from './Card'

describe('Card', () => {
  it('should render', () => {
    const wrapper = create(
      <BrandProvider>
        <Card>TEST</Card>
      </BrandProvider>,
    )
    const serialized = wrapper.toJSON()
    expect(serialized).toMatchSnapshot()
  })
})
