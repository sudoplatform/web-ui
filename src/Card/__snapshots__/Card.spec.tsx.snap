// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
.c0 {
  width: 100%;
  border-radius: 4px;
  box-shadow: rgba(0,0,0,0.12) 0px 2px 10px 0px;
  padding: 20px;
  background: white;
}

<div
  className="c0"
>
  TEST
</div>
`;
